src/main/kotlin/com/pubnub/api/crypto/cryptor/LegacyCryptor.kt

```diff
@@ -57,28 +62,57 @@ class LegacyCryptor(val cipherKey: String, val useRandomIv: Boolean = true) : Cr
         }
     }
 
-    override fun encryptStream(stream: InputStream): Result<EncryptedStreamData> {
-        TODO("Not yet implemented")
+    override fun encryptStream(stream: InputStream): EncryptedStreamData {
+        try {
+            val ivBytes: ByteArray = getIvBytesForEncryption()
+            val ivSpec: AlgorithmParameterSpec = IvParameterSpec(ivBytes)
+            val keyBytes: ByteArray = getKeyAsShaInHexBytes(cipherKey)
+            val newKey = SecretKeySpec(keyBytes, "AES")
+            val cipher = Cipher.getInstance(CIPHER_TRANSFORMATION)
+            cipher.init(Cipher.ENCRYPT_MODE, newKey, ivSpec)
+            val cipheredStream = CipherInputStream(stream, cipher)
+            return EncryptedStreamData(stream = SequenceInputStream(ivBytes.inputStream(), cipheredStream))
+        } catch (e: Exception) {
+            throw PubNubException(errorMessage = e.message, pubnubError = PubNubError.CRYPTO_ERROR)
+        }
     }
 
-    override fun decryptStream(encryptedData: EncryptedStreamData): Result<InputStream> {
-        TODO("Not yet implemented")
+    override fun decryptStream(encryptedData: EncryptedStreamData): InputStream {
+        try {
+
+            val cipher = Cipher.getInstance(CIPHER_TRANSFORMATION)
+            val ivBytes = ByteArray(cipher.blockSize)
+            val numberOfReadBytes = encryptedData.stream.read(ivBytes)
+            if (numberOfReadBytes != cipher.blockSize) {
+                throw PubNubException(
+                    errorMessage = "Could not read IV from encrypted stream",
+                    pubnubError = PubNubError.CRYPTO_ERROR
+                )
+            }
+            val ivSpec: AlgorithmParameterSpec = IvParameterSpec(ivBytes)
+            val keyBytes: ByteArray = getKeyAsShaInHexBytes(cipherKey)
+            val newKey = SecretKeySpec(keyBytes, "AES")
+            cipher.init(Cipher.DECRYPT_MODE, newKey, ivSpec)
+            return CipherInputStream(encryptedData.stream, cipher)
+        } catch (e: Exception) {
+            throw PubNubException(errorMessage = e.message, pubnubError = PubNubError.CRYPTO_ERROR)
+        }
     }
 
     private fun getKeyAsShaInHexBytes(cipherKey: String): ByteArray {
-        val cipherKeyBytes = cipherKey.toByteArray(Charsets.UTF_8)
-        val cipherKeySha256Bytes: ByteArray = MessageDigest.getInstance("SHA-256").digest(cipherKeyBytes)
-        val hexString = cipherKeySha256Bytes.joinToString("") { "%02x".format(it) }
-        return hexString.toByteArray(Charsets.UTF_8)
+        return String(
+            Crypto.hexEncode(Crypto.sha256(cipherKey.toByteArray(charset(ENCODING_UTF_8)))),
```

Comment:
I assume that this is temporary solution and we want to replace Crypto.java with equivalent code in kotlin?

Reply:
I move the methods to cryptors